### PR TITLE
relax some dependencies from `<=` to `<`

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

to prevent very minor bug-fix version bumps

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

File: requirements/app/app.txt [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
 lightning-utilities >=0.8.0, <0.10.0
-beautifulsoup4 >=4.8.0, <4.12.3
-inquirer >=2.10.0, <=3.1.3
+beautifulsoup4 >=4.8.0, <4.13.0
+inquirer >=2.10.0, <=3.2.0
 psutil <5.9.6
-click <=8.1.6
-python-multipart>=0.0.5, <=0.0.6
+click <8.2
+python-multipar t>=0.0.5, <=0.0.6
```
Comment (Borda) [inline requirements/app/app.txt:15]:
```suggestion
python-multipart >=0.0.5, <=0.0.6
```

File: requirements/app/app.txt [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
+fsspec >=2022.5.0, <2023.7.0
 croniter >=1.3.0, <1.5.0  # strict; TODO: for now until we find something more robust.
 traitlets >=5.3.0, <5.10.0
-arrow >=1.2.0, <1.2.4
+arrow >=1.2.0, <1.3.0
 lightning-utilities >=0.8.0, <0.10.0
-beautifulsoup4 >=4.8.0, <4.12.3
-inquirer >=2.10.0, <=3.1.3
+beautifulsoup4 >=4.8.0, <4.13.0
+inquirer >=2.10.0, <=3.2.0
```
Comment (carmocca) [inline requirements/app/app.txt:12]: Why have you left some dependencies with "<="? Don't you want to update all?

File: requirements/app/test.txt [hunk 3]
```diff
@@ -11,6 +11,6 @@ playwright ==1.36.0
 httpx ==0.24.1
 trio <0.22.0  # strict https://github.com/python-trio/trio/pull/2213
 pympler
-psutil <=5.9.5
+psutil <5.6.0
```
Comment (carmocca) [inline requirements/app/test.txt:14]: This is a downgrade. Is this intended?

**Review by carmocca — APPROVED** [review]

File: requirements/app/app.txt (hunk 2 above)
Comment (Borda) [reply requirements/app/app.txt:12]:
```suggestion
inquirer >=2.10.0, <3.2.0
```

File: requirements/app/test.txt (hunk 3 above)
Comment (Borda) [reply requirements/app/test.txt:14]:
```suggestion
psutil <5.10.0
```

**Review by awaelchli — APPROVED** [review]
I'm ok trying this. But if our development gets blocked by package releases influencing our CI, we should consider reverting back to the more stricter version limits

**Pull request merged**